### PR TITLE
Development docs

### PR DESCRIPTION
**Pull request by bklaas**
I decided to use my last day on the bench (!) to try to draft up a new round of development documentation for Charlie.

Highlights:
* I took an opinionated approach to the docs. A Charlie dev will use 
  * Docker to lint and unit test
  * cloud.gov for testing deployment
  * connect to `GSA TTS Testing` Slack for confirmation testing
* I separated developer docs from the README to modularize things a bit.

It is ENTIRELY possible I have some details wrong. I'm happy to iterate on this until it's deemed accurate and functional. Thanks for the review!

### CODE REVIEW

**Review by jskinne3 — APPROVED** [review]
These steps make sense to me. One question about the `DEV.md` file: Would it be more conventional to put this in a `CONTRIBUTING.md` file?

**Comment by bklaas** [comment]
That's a good idea. I'll make the change to fold DEV.md into CONTRIBUTING.md today.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,35 +1,134 @@
...
+
+#### Getting the New Bot Deployed to Production
+
+After `awesomobot` has been linted, unit tested, live tested on `GSA TTS Testing` Slack, and deemed production-ready, the process to get the new bot to `TTS` Slack is as follows:
+
+- Push your `awesomo` branch to github.com/18F/Charlie
+- File a Pull Request for `awesomo->main`
+  -- PR should be from your development
+  -- Include an explanatory paragraph in the PR as to what's in the request.
+- Request a reviewer. If you don't know who should review the code, ask in the [#bots](https://app.slack.com/client/T025AQGAN/C02FPFGBG) channel
```
Comment (mgwalker) [inline CONTRIBUTING.md:123]:
```suggestion
- Get your PR reviewed. By default @18F/charlie-maintainers will be assigned, and you can, ask in the [#bots](https://app.slack.com/client/T025AQGAN/C02FPFGBG) channel for anyone else to review
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,35 +1,134 @@
...
+Note: it takes a few minutes for the code to fully build and deploy. The console should give updates of its progress and exit to a prompt when finished.
+
+#### Getting the New Bot Deployed to Production
+
+After `awesomobot` has been linted, unit tested, live tested on `GSA TTS Testing` Slack, and deemed production-ready, the process to get the new bot to `TTS` Slack is as follows:
+
+- Push your `awesomo` branch to github.com/18F/Charlie
+- File a Pull Request for `awesomo->main`
+  -- PR should be from your development
+  -- Include an explanatory paragraph in the PR as to what's in the request.
```
Comment (mgwalker) [inline CONTRIBUTING.md:122]:
```suggestion
   - PR should be from your development
   - Include an explanatory paragraph in the PR as to what's in the request.
```

Second-level lists in GH Markdown require just indentation. The `--` doesn't trigger list items, so the rendering goes wonky.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,35 +1,134 @@
...
+- Request a reviewer. If you don't know who should review the code, ask in the [#bots](https://app.slack.com/client/T025AQGAN/C02FPFGBG) channel
+- After PR has passed review, merge the changes to `main` (sometimes the reviewer does this)
+  -- CI/CD is setup so the deployment to `TTS` Slack is automatic
```
Comment (mgwalker) [inline CONTRIBUTING.md:125]:
```suggestion
   - CI/CD is setup so the deployment to `TTS` Slack is automatic
```

**Review by mgwalker — APPROVED** [review]
This is really helpful! I added some notes and a couple of minor suggestions. Feel free to take them or leave them.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,35 +1,134 @@
...
 
-### Before you make a pull request
+## Development
 
-Please ensure the code linter and tests pass.
+Charlie is a Slack App, and for structural and policy reasons described below, it can't be fully qualified and exercised in a local development environment.
+
+Because Slack apps work by responding to web hooks rather than setting up a persistent websocket connection, the machine where Charlie runs cannot respond to Slack events unless it is accessible on the public internet over HTTPS. **GSA policy forbids our computers from doing this**. As a result, it is necessary to run the app in cloud.gov to test it.
+
+To facilitate this requirement, there is a `GSA TTS Testing` Slack instance available to test in-development changes to Charlie before deploying to production. The Slack instance `GSA TTS Testing` is deployed in cloud.gov and this github repository is fully wired up with CI/CD to automatically deploy to it (for branches other than `main`).
```
[the remark this answers is not in the document]
Comment (bklaas) [reply CONTRIBUTING.md:17]: When I built the q-expander bot I didn't have to do anything other than `cf push` and wait a few minutes to have that code operational in `GSA TTS Testing` Slack. There has to be some kind of CI/CD glued up, doesn't there? I do think I have something not quite right though...is Slack even involved in any part of the CI/CD bits?

[the remark this answers is not in the document]
Comment (mgwalker) [reply CONTRIBUTING.md:17]: CI/CD does publish to production on merges to `main`, but it doesn't do anything with the test Slack instance. We could fix that!

File: CONTRIBUTING.md (hunk 2 above)
Comment (bklaas) [reply CONTRIBUTING.md:122]: Agreed. Believe it or not it's the VS Code auto-formatter that made that change.

File: CONTRIBUTING.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (bklaas) [reply CONTRIBUTING.md:17]: I'm confused...how were my changes on dev branches automatically functional in `GSA TTS Testing` Slack? I don't know of anything I did other than `cf push` to get things working on that Slack.

[the remark this answers is not in the document]
Comment (mgwalker) [reply CONTRIBUTING.md:17]: I suspect I'm not understanding what you mean, so here's where my head is. When I read this line:

> this github repository is fully wired up with CI/CD to automatically deploy to it

It makes me think that doing a `git push` to my feature branch will result in a new deployment into the `gsa-18f-hubot/dev` space in cloud.gov, which is not the case. You have to run `cf push` from your local machine, which is not automatic.

**Pull request merged**